lib/text_recognition/save/widget.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:readnod/navigation.dart';
+import 'package:readnod/text_recognition/camera/widget.dart';
+import 'package:readnod/text_recognition/save/bloc.dart';
+import 'package:readnod/text_recognition/save/events.dart';
+import 'package:readnod/text_recognition/save/states.dart';
+import 'package:readnod/translations.dart';
+
+class SaveArguments {
+  final String textToShare;
+
+  SaveArguments({@required this.textToShare});
+}
+
+class SaveWidget extends StatefulWidget {
+  static final route = "/text/recognition/save";
+
+  @override
+  _SaveWidgetState createState() => _SaveWidgetState();
+}
+
+class _SaveWidgetState extends State<SaveWidget> {
+  SaveBloc _bloc;
+  final TextEditingController _textController = TextEditingController();
+  final FocusNode _textFocus = FocusNode();
+
+  @override
+  Widget build(BuildContext context) {
+    _bloc = SaveBloc(repository: RepositoryProvider.of(context));
```

Comment:
consider using BlocProvider to manage blocs  https://github.com/felangel/bloc/blob/e818f90e5b08688f3ddfe9ea297d10be8a9249e3/docs/_snippets/flutter_firestore_todos_tutorial/main.dart.md